bundles/org.openhab.binding.sonnen/src/main/java/org/openhab/binding/sonnen/internal/SonnenHandler.java

```diff
@@ -0,0 +1,276 @@
+        }
+    }
+
+    private void updateLinkedChannels() {
+        verifyLinkedChannel(CHANNELBATTERYCHARGING);
+        verifyLinkedChannel(CHANNELBATTERYDISCHARGING);
+        verifyLinkedChannel(CHANNELGRIDFEEDIN);
+        verifyLinkedChannel(CHANNELCONSUMPTION);
+        verifyLinkedChannel(CHANNELSOLARPRODUCTION);
+        verifyLinkedChannel(CHANNELBATTERYLEVEL);
+        verifyLinkedChannel(CHANNELFLOWCONSUMPTIONBATTERY);
+        verifyLinkedChannel(CHANNELFLOWCONSUMPTIONGRID);
+        verifyLinkedChannel(CHANNELFLOWCONSUMPTIONPRODUCTION);
+        verifyLinkedChannel(CHANNELFLOWGRIDBATTERY);
+        verifyLinkedChannel(CHANNELFLOWPRODUCTIONBATTERY);
+        verifyLinkedChannel(CHANNELFLOWPRODUCTIONGRID);
+
+        if (!linkedChannels.isEmpty()) {
+            updatebatteryData(null);
+            for (Channel channel : getThing().getChannels()) {
+                updateChannel(channel.getUID().getId());
+            }
+            startAutomaticRefresh();
+            automaticRefreshing = true;
+        }
+    }
+
+    private void verifyLinkedChannel(String channelID) {
+        if (isLinked(channelID) && !linkedChannels.containsKey(channelID)) {
+            linkedChannels.put(channelID, true);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        stopScheduler();
+    }
+
+    private void stopScheduler() {
+        ScheduledFuture<?> job = refreshJob;
+        if (job != null) {
+            job.cancel(true);
+        }
+        refreshJob = null;
+    }
+
+    /**
+     * Start the job refreshing the oven status
+     */
+    private void startAutomaticRefresh() {
+        ScheduledFuture<?> job = refreshJob;
+        if (job == null || job.isCancelled()) {
+            int period = config.refreshInterval;
+            refreshJob = scheduler.scheduleWithFixedDelay(this::run, 0, period, TimeUnit.SECONDS);
+        }
+    }
+
+    private void run() {
+        updatebatteryData(null);
+        for (Channel channel : getThing().getChannels()) {
+            updateChannel(channel.getUID().getId());
+        }
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        if (!automaticRefreshing) {
+            logger.debug("Start automatic refreshing");
+            startAutomaticRefresh();
+            automaticRefreshing = true;
+        }
+        verifyLinkedChannel(channelUID.getId());
+        updateChannel(channelUID.getId());
+    }
+
+    @Override
+    public void channelUnlinked(ChannelUID channelUID) {
+        linkedChannels.remove(channelUID.getId());
+        if (linkedChannels.isEmpty()) {
+            automaticRefreshing = false;
+            stopScheduler();
+            logger.debug("Stop automatic refreshing");
+        }
+    }
+
+    private void updateChannel(String channelId) {
+        if (isLinked(channelId)) {
+            State state = null;
+            SonnenJsonDataDTO data = serviceCommunication.getBatteryData();
+            if (data != null) {
```

Comment:
As previously mentioned, as you implemented it, my feeling is that data cannot be null.
Because you are building a default object is the HTTP request or its desierialization is failing.

Reply:
In fact, this might be the good approach to have the data being null in case the request failed.

Reply:
In general I agree, but If I dont put it I get possible null pointer warning from eclipse and if i run spotless:apply. Therefore I kept it

Reply:
To be clarified but I do not understand why spotless could be involved here...

Reply:
I mean I get findings in the code-analysis report

Reply:
It is ok to keep this test.